controller/src/main.rs

```diff
@@ -70,16 +82,23 @@ async fn main() -> Result<()> {
     .context(controller_error::PrometheusServerError)?
     .run();
 
-    // TODO if any of these fails, we should write to the k8s termination log and exit.
     tokio::select! {
-        _ = drainer => {
-            event!(Level::ERROR, "reflector drained");
+        _ = brs_drainer => {
+            event!(Level::ERROR, "brs reflector drained");
+            return Err(controller_error::Error::KubernetesWatcherFailed {object: "brs".to_string()});
+        },
+        _ = node_drainer => {
+            event!(Level::ERROR, "node reflector drained");
+            return Err(controller_error::Error::KubernetesWatcherFailed {object: "node".to_string()});
```

Comment:
Any reason returning Err here? The Result<()> in main is actually not parsed by any other handler. If this is for log purpose, does it make sense to log in the `event!` rather than return Err?